### PR TITLE
Add docs for QTAIM method

### PR DESCRIPTION
**Pull request by mscho527**
_Related Issue: #886_

This PR adds documentations on how to use Bader's QTAIM method in `cclib`.

### CODE REVIEW

File: doc/sphinx/methods.rst [hunk 1]
```diff
@@ -306,6 +306,61 @@ Notes
...
 
+.. index::
+    single: methods; Bader's QTAIM
+    
+Bader's QTAIM
+----------------
+
+Bader's QTAIM charges define the border between the atoms in the molecule as a surface where each point on the surface has zero flux. In other words, the points on the surface of the Bader spaces satisfy the equation :math:`\nabla \rho (r) \cdot n(r) = 0`. In cclib, numerical calculation of QTAIM charges through the algorithm proposed in `[Henkelman2006] <http://theory.cm.utexas.edu/henkelman/code/bader/download/henkelman06_354.pdf>`_  is implemented.
+
+Calculating the Bader's QTAIM charges in cclib follow similar steps as other population analysis methods. Following code provides an example of how QTAIM charges can be obtained.
```
Comment (shivupa) [inline doc/sphinx/methods.rst:317]: "The following code...."

File: doc/sphinx/methods.rst [hunk 2]
```diff
@@ -306,6 +306,61 @@ Notes
...
+    # Inputs for Volume object below are origin, top corner, and spacing
+    # represented in Cartesian coordinates and in angstroms.
+    vol = Volume((-3, -3, -3), (3, 3, 3), (.1, .1, .1))
+
+    m = Bader(d, vol)
+    m.calculate()
+
+After the calculate() method is called, the following attributes are available:
+
+* ``fragresults``: a three dimensional array x, y, and z coordinates from the Volume object as the axes, so that ``fragresults[1][2][3]`` gives the Bader space (in integers starting from 1) that the grid space in (0, 1, 2) position is classified into.
```
Comment (shivupa) [inline doc/sphinx/methods.rst:339]: "is classified into" -> "belongs to"

File: doc/sphinx/methods.rst [hunk 3]
```diff
@@ -306,6 +306,61 @@ Notes
...
+* ``matches``: a vector with the Bader space (integers starting from 1) that an atom is matched with.
+* ``fragcharges``: a vector with the number of (partial) electrons in each atom, so that ``fragcharges[2]`` gives the number of electrons in the 3rd atom.
```
Comment (shivupa) [inline doc/sphinx/methods.rst:341]: is this in the third bader space or actually the third atom?

File: doc/sphinx/methods.rst [hunk 4]
```diff
@@ -306,6 +306,61 @@ Notes
...
+
+Since some computational chemistry packages support writing out charge densities as cube files, it may be desirable to use the cube files that were prepare previously to calculate Bader's QTAIM charges.
+To do so, ``Volume`` object should be prepared by reading in from the cube file and should be passed into Bader object as an input as in the following code:
```
Comment (shivupa) [inline doc/sphinx/methods.rst:344]: To do so, a ``Volume`` object should be prepared by reading in a cube file and should be passed into a Bader object as shown below:

File: doc/sphinx/methods.rst (hunk 1 above)
Comment (mscho527) [reply doc/sphinx/methods.rst:317]: I've corrected this. Thanks!

File: doc/sphinx/methods.rst (hunk 2 above)
Comment (mscho527) [reply doc/sphinx/methods.rst:339]: I've fixed this. Thanks!

File: doc/sphinx/methods.rst (hunk 3 above)
Comment (mscho527) [reply doc/sphinx/methods.rst:341]: It's third atom (as in the order of the atoms in ccData object's `atomcoords`.)

File: doc/sphinx/methods.rst [hunk 5]
```diff
@@ -306,6 +306,61 @@ Notes
...
+    m = Bader(d, vol)
+    m.calculate()
+
+After the calculate() method is called, the following attributes are available:
+
+* ``fragresults``: a three dimensional array x, y, and z coordinates from the Volume object as the axes, so that ``fragresults[1][2][3]`` gives the Bader space (in integers starting from 1) that the grid space in (0, 1, 2) position belongs to.
+* ``matches``: a vector with the Bader space (integers starting from 1) that an atom is matched with.
+* ``fragcharges``: a vector with the number of (partial) electrons in each atom, so that ``fragcharges[2]`` gives the number of electrons in the 3rd atom.
+
+Since some computational chemistry packages support writing out charge densities as cube files, it may be desirable to use the cube files that were prepare previously to calculate Bader's QTAIM charges.
```
Comment (shivupa) [inline doc/sphinx/methods.rst:343]: We might want to say that this would be considerably faster and is recommended for larger systems?

Comment (mscho527) [reply doc/sphinx/methods.rst]: That's true :) I've updated this line to emphasize this. Thanks!

**Pull request merged**